### PR TITLE
[DEVP-1023] Update diff --git and --with-region to show full diff whe…

### PR DESCRIPTION
**Pull request by gleaver**
…n a service added or removed. Also bail on loading a manifest from a region where it does not exist.

Catch an edge case where if services are added or removed we fail to diff them correctly. This happens because we consider a filebacked manifest to be valid even if the region it is loaded for is not used.

This change bails if attempting to load a manifest for an unused region, and amends the CRD diff functionality to consider the lack of a region to be an empty string.

Feedback appreciated on ways in which this might break things/not behave as expected.

### CODE REVIEW

**Comment by gleaver** [comment]
Altered implementation to only verify the region is correct in the diff functionality, as verifying during load seemed too low-level a place to put the bail.

**Comment by gleaver** [comment]
Related manifests change: https://github.com/Babylonpartners/manifests/pull/14982

File: shipcat_cli/src/diff.rs [hunk 1]
```diff
@@ -8,15 +8,30 @@ use super::{Config, Region, Manifest, Result};
...
+/// otherwise YAML serialise the content. For diff purposes, the content
+/// of a manifest not in a region is a blank, rather than being invalid.
+fn as_yaml(svc: &str, conf: &Config, region: &Region) -> Result<String> {
+    let mf = shipcat_filebacked::load_manifest(&svc, conf, region)?;
+    if let Ok(m) = mf.verify_region() {
+        let yaml = serde_yaml::to_string(&m)?;
+        Ok(yaml)
+    } else {
+        Ok("".to_string())
+    }
```
Comment (clux) [inline shipcat_cli/src/diff.rs]: wait `load_manifest` works in a region that it's not included in?
i guess that's part of the #300 bug?

Comment (gleaver) [reply shipcat_cli/src/diff.rs]: That's the bug, yeah.

**Pull request merged**